File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let countary = new Array(1001).fill(0);
    console.log(countary)
    for (let i=0; i<array.length; i++){
        countary[array[i]]++; // array를 인덱스로 보기
    }
    
    let max = 0;  // 나온 횟수
    let mode = 0; // 최빈값
    
    let count = 0;  // 최빈값이 중복되는지 확인하는 변수
    
    console.log(countary);
    
    for (let i =0; i<countary.length; i++){
        if (max<countary[i]){
            max = countary[i];
            mode = i; 
        }
    }
    
    console.log('max = ' + max);
    console.log('mode = '+mode);
   
    for (let i =0; i<countary.length; i++){
        if (countary[i] == max) {count++}
    }
    
    console.log('count = '+count);
    if (count > 1 ){return -1}
    else return mode;
}

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// array_len은 배열 array의 길이입니다.
int solution(int array[], size_t array_len) {
    int count[1001] = {0, };
    int max = 0; // 나온 횟수
    int mode = 0; // 최빈값
    int cnt = 0; // 최빈값이 몇 개인지 확인
       
    for (int i=0; i<array_len; i++)
    {
        count[array[i]]++;
    }
    for (int i=0; i<1001; i++){
        if(count[i] > max ){
            max = count[i];
            
            mode = i;
        }
    }
    
    for (int i= 0; i<1001; i++){
        if (count[i] == max) cnt++;
    }
    
    if (cnt>1) return -1;
    else return mode;
}